leo/main.rs

```diff
@@ -14,80 +14,198 @@
-            LogoutCommand::new().display_order(16),
-        ])
-        .set_term_width(0);
-
-    let mut help = app.clone();
-    let arguments = app.get_matches();
-
-    match arguments.subcommand() {
-        ("new", Some(arguments)) => NewCommand::process(arguments),
-        ("init", Some(arguments)) => InitCommand::process(arguments),
-        ("build", Some(arguments)) => BuildCommand::process(arguments),
-        ("watch", Some(arguments)) => WatchCommand::process(arguments),
-        ("test", Some(arguments)) => TestCommand::process(arguments),
-        ("setup", Some(arguments)) => SetupCommand::process(arguments),
-        ("prove", Some(arguments)) => ProveCommand::process(arguments),
-        ("run", Some(arguments)) => RunCommand::process(arguments),
-        ("login", Some(arguments)) => LoginCommand::process(arguments),
-        ("add", Some(arguments)) => AddCommand::process(arguments),
-        ("remove", Some(arguments)) => RemoveCommand::process(arguments),
-        ("publish", Some(arguments)) => PublishCommand::process(arguments),
-        ("deploy", Some(arguments)) => DeployCommand::process(arguments),
-        ("clean", Some(arguments)) => CleanCommand::process(arguments),
-        ("lint", Some(arguments)) => LintCommand::process(arguments),
-        ("update", Some(arguments)) => UpdateCommand::process(arguments),
-        ("logout", Some(arguments)) => LogoutCommand::process(arguments),
-        _ => {
-            // Set logging environment
-            match arguments.is_present("debug") {
-                true => logger::init_logger("leo", 2),
-                false => logger::init_logger("leo", 1),
-            }
-
-            Updater::print_cli();
-
-            help.print_help()?;
-            println!();
-            Ok(())
+pub mod api;
+pub mod commands;
+pub mod config;
+pub mod context;
+pub mod logger;
+pub mod synthesizer;
+pub mod updater;
+
+use anyhow::Error;
+use std::process::exit;
+
+use commands::{
+    package::{Add, Login, Logout, Publish, Remove},
+    Build,
+    Clean,
+    Command,
+    Deploy,
+    Init,
+    Lint,
+    New,
+    Prove,
+    Run,
+    Setup,
+    Test,
+    Update,
+    Watch,
+};
+
+use structopt::{clap::AppSettings, StructOpt};
+
+/// CLI Arguments entry point - includes global parameters and subcommands
+#[derive(StructOpt, Debug)]
+#[structopt(name = "leo", author = "The Aleo Team <hello@aleo.org>", setting = AppSettings::ColoredHelp)]
+struct Opt {
+    #[structopt(short, long, help = "Print additional information for debugging")]
+    debug: bool,
+
+    #[structopt(short, long, help = "Suppress CLI output")]
+    quiet: bool,
+
+    #[structopt(subcommand)]
+    command: CommandOpts,
+}
+
+///Leo compiler and package manager
+#[derive(StructOpt, Debug)]
+#[structopt(setting = AppSettings::ColoredHelp)]
+enum CommandOpts {
+    #[structopt(about = "Init new Leo project command in current directory")]
+    Init {
+        #[structopt(flatten)]
+        cmd: Init,
```

Comment:
1. nit: please rename `cmd` to `command`
2. question: if we're telling `structopt` to simply flatten the command, is there a reason we need to initialize the inner `cmd`?

Reply:
1. Done.
2. We need to, yes. [Flattening allows using external structs as subcommands](https://docs.rs/structopt/0.3.21/structopt/#flattening-subcommands) + see how this code is called within main function.